src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -484,6 +964,118 @@ get_services_properties(sd_bus_message *m, const char *path)
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "Immutable")) {
+            bool immutable;
+
+            r = sd_bus_message_enter_container(m, SD_BUS_TYPE_VARIANT, "b");
+            SOL_INT_CHECK(r, < 0, r);
+
+            r = sd_bus_message_read_basic(m, 'b', &immutable);
+            SOL_INT_CHECK(r, < 0, r);
+
+            service->immutable = immutable;
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "AutoConnect")) {
+            bool autoconnect;
+
+            r = sd_bus_message_enter_container(m, SD_BUS_TYPE_VARIANT, "b");
+            SOL_INT_CHECK(r, < 0, r);
+
+            r = sd_bus_message_read_basic(m, 'b', &autoconnect);
+            SOL_INT_CHECK(r, < 0, r);
+
+            service->autoconnect = autoconnect;
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "Roaming")) {
+            bool roaming;
+
+            r = sd_bus_message_enter_container(m, SD_BUS_TYPE_VARIANT, "b");
+            SOL_INT_CHECK(r, < 0, r);
+
+            r = sd_bus_message_read_basic(m, 'b', &roaming);
+            SOL_INT_CHECK(r, < 0, r);
+
+            service->roaming = roaming;
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "Nameservers")) {
+            const char *nameserver = NULL;
+
+            sol_ptr_vector_clear(&service->nameservers);
+
+            r = sd_bus_message_enter_container(m, 'a', "s");
+            SOL_INT_CHECK(r, < 0, r);
+
+            while (sd_bus_message_read_basic(m, 's', &nameserver) > 0) {
+                r = sol_ptr_vector_append(&service->nameservers, nameserver);
+                SOL_INT_CHECK(r, < 0, r);
+            }
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "Timeservers")) {
+            const char *timeserver = NULL;
+
+            sol_ptr_vector_clear(&service->timeservers);
+
+            r = sd_bus_message_enter_container(m, 'a', "s");
+            SOL_INT_CHECK(r, < 0, r);
+
+            while (sd_bus_message_read_basic(m, 's', &timeserver) > 0) {
+                r = sol_ptr_vector_append(&service->timeservers, timeserver);
+                SOL_INT_CHECK(r, < 0, r);
+            }
+
+            r = sd_bus_message_exit_container(m);
+            SOL_INT_CHECK(r, < 0, r);
+        } else if (streq(str, "Domains")) {
+            const char *domain = NULL;
+
+            sol_ptr_vector_clear(&service->domains);
+
+            r = sd_bus_message_enter_container(m, 'a', "s");
+            SOL_INT_CHECK(r, < 0, r);
+
+            while (sd_bus_message_read_basic(m, 's', &domain) > 0) {
+                r = sol_ptr_vector_append(&service->domains, domain);
+                SOL_INT_CHECK(r, < 0, r);
```

Comment:
You should exit the container and you will leave the nameservers. I think you should iterate over the list the clear all the elements and them clear the list itself.